lib/consul/async/consul_template.rb

```diff
@@ -188,6 +188,11 @@ def kv(name = nil, dc: nil, keys: nil, recurse: false)
         create_if_missing(path, query_params) { ConsulTemplateKV.new(ConsulEndpoint.new(consul_conf, path, true, query_params, default_value), name) }
       end
 
+      def json(url)
```

Comment:
I am now trying to avoid polluting too much the base namespace because it can conflict with variables.

Can you please namespace your function in an object as done here:
https://github.com/criteo/consul-templaterb/blob/master/lib/consul/async/consul_template.rb#L25

Example of name of object `remote_resouce` with as function `json()` or `as_json` (since we might add other functions such as raw() or xml() or whatever...

Reply:
Another issue, that would be nice to provide a mandatory default value (`[]` or `{}`), so rendering of template can start at first second. Since JSON can be array or object, it would ease quite a lot the rendering of templates:

```
remote_resource.json(url, '[]', refresh_delay_secs: 30).each do |obj|
%><%= obj['Name'] %>
<%
end
```
By providing '[]' => the rendering is done at startup and does not require 'IF' everywhere.

Reply:
done for the namespacing and default value.